### PR TITLE
Fix +- in complex numbers and set mpl to all math minus signs

### LINKED ISSUE #4857: mpl drawer has inconsistent use of hyphens and math minus signs
### Information

- **Qiskit Terra version**: master
- **Python version**: 3.8
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

The mpl drawer changes leading hyphens in parameters to the math mode minus sign, which is similar in width to the + and about twice the width of a hyphen. There are other places, such as e notation and the imaginary part of complex numbers, where a minus is intended, but a hyphen is used. Note below.

### Steps to reproduce the problem
```
from qiskit import QuantumCircuit
qc = QuantumCircuit(1)
qc.u1(complex(-1.0, -1.0), 0)
qc.u1(-1e-7, 0)
qc.draw('mpl')
```
image
Also note that pi_check returns +- for the imaginary part instead of just -.

### Suggested solutions
For all parameter expressions in the mpl drawer, convert hyphens to minus signs. Since all parameters go through pi_check, this could be done at the pi_check level.

### PR DESCRIPTION
**Pull request by enavarro51**
### Summary

Fixes #4857

### Details and comments

For the text and mpl drawers, this PR changes the output of pi_check for complex numbers from `1.0+-1.0j` to `1.0-1.0j`, and for the mpl drawer, converts all param hyphens to the math mode minus.

### CODE REVIEW

**Comment by enavarro51** [comment]
I think a text drawer display change is outside the scope of this issue and PR. As for the latex issue, I believe switching to `\pi\jmath` will solve the problem.

**Comment by enavarro51** [comment]
@Cryoris In regards to the text drawer display of pi, I think using `π` throughout would be nice and there is a pi in the cp437 encoding. I just wonder if there should be a discussion of this since it shows in lots of circuits out there, and there also might be spacing issues to address since it's now 1 char instead of 2.

**Comment by 1ucian0** [comment]
> @Cryoris In regards to the text drawer display of pi, I think using `π` throughout would be nice and there is a pi in the cp437 encoding. I just wonder if there should be a discussion of this since it shows in lots of circuits out there, and there also might be spacing issues to address since it's now 1 char instead of 2.

I'm thinking about getting out of cp437. Can we have a 'utf8' in `pi_check` to allow `π`? I need to land https://github.com/Qiskit/qiskit-terra/pull/4900 before support multiple encodings depending on the env.

**Comment by enavarro51** [comment]
Two things. First `√` for the sqrt-X is part of cp437 and works fine in the text drawer and `π `is also in cp437. So I don't think changing  to utf-8 is a requirement, though it may be a good idea for other reasons.

Secondly, it looks like 24 tests will have to be modified if we do this, so I'm wondering if it makes sense to make this a new issue?

File: qiskit/circuit/tools/pi_check.py [hunk 1]
```diff
@@ -135,8 +135,15 @@ def normalize(single_inpt):
     complex_inpt = complex(inpt)
     real, imag = map(normalize, [complex_inpt.real, complex_inpt.imag])
 
+    jstr = '\\jmath' if output == 'latex' else 'j'
```
Comment (Cryoris) [inline qiskit/circuit/tools/pi_check.py:138]: This should also cover latex_source
```suggestion
    jstr = '\\jmath' if output in ['latex', 'latex_source'] else 'j'
```

**Review by Cryoris — APPROVED** [review]
LGTM, can you open another issue for the display of pi?

**Pull request merged**